Sources/StreamChat/Query/Filter.swift

```diff
@@ -111,23 +113,47 @@ public struct Filter<Scope: FilterScope> {
     ///   - key: The "left-hand" side of the filter. Specifies the name of the field the filter should match.
     ///   - value: The "right-hand" side of the filter. Specifies the value the filter should match.
     ///
-    public init(operator: String, key: String?, value: FilterValue) {
+    public init(
+        operator: String,
+        key: String?,
+        value: FilterValue,
+        keyPathValueProvider: (() -> String)? = nil
```

Comment:
We should add init docs for the the new argument `keyPathValueProvider` 👍

Reply:
I moved a little bit the parts here in order to hide the keypath and not allow the customers to pass in a value (for now at least). Let me know your thoughts